### ISSUE TITLE
Check if accessibility regressions exist compared to old DevTools

### ISSUE THREAD

**Issue by gaearon** [bvaughn/react-devtools-experimental#52]
Before this becomes stable, we need to check if we are regressing accessibility on any important existing interactions.

At least, we should probably make the tree view focusable.

**Comment by bvaughn** [comment]
One example that would be nice to improve (but I think might be trickier than you would assume) is keyboard focus and navigation within the Components tree.

**Comment by bvaughn** [comment]
c60f991 and d1a5346 should significantly improve keyboard accessibility for the main "Components" tab. It's probably still pretty rough in the Profiler tab though.

How sure how to support keyboard navigation of the flamegraph. "chrome://tracing" uses WASD keys to zoom in and out, but its more of an analog zoom and I think that would require a significant change for DevTools to operate that way.

**Comment by bvaughn** [comment]
Also commit https://react-devtools-experimental-ztikpbkpml.now.sh

**Comment by AlmeroSteyn** [comment]
Hi, I love the look and feel of the new dev tools.

Been looking a bit at a11y regressions on the dev tools. Purely looking at regressions from the old dev tools I can currently find only one issue while testing with keyboard only as well as screen reader. I have not tested with other AT such as voice control or other input methods such as switches etc.

I tested with:
Windows 10
Firefox 66.0.3
Chrome 74.0.3729.131
Chrome Canary 76.0.3782.0
NVDA 2019.1.1

NVDA accounts for more than 30% of screen reader use.

Keyboard a11y has seen a great improvement in the new dev tools. Awesome work!

In the old dev tools I am unable to reach the Profiler sub tab or anything but the main bar buttons. This is also the reason for a single regression as almost the entire old devtools is keyboard inaccessible, hiding everything inside from inspection.

**The issue**

Button texts are no longer read out in NVDA screen reader. All controlling buttons are simply read out as `button`. In the old devtools all reachable buttons still have text read out by NVDA.

I see that the buttons carry an `aria-labelledby` property which points to the tooltips. However it is usually recommended that the elements decorated with the `id` values being referenced by `aria-labelledby` and `aria-describedby` exists in the rendered DOM if these properties are applied. 

The current state where they refer to `id` values even when they are not rendered will cause issues in some screen readers, NVDA being one. This has to do with how the accessibility tree is resolved by the screen reader and how it builds its own object model.

Suggestion would to provide visually hidden button texts or to use the `aria-label` property instead. In both cases it would mean duplication of the tooltip text so the tooltips themselves would then require `aria-hidden="true"`. 

Alternatively one could also keep the tooltip rendered and apply a visually hidden style. This would mean that screen reader users could encounter these elements by mistake but being rendered at the base of the DOM it is acceptable.

**Comment by bvaughn** [comment]
Whoa, thanks for the helpful info and research @AlmeroSteyn!

cc @ryanflorence about the tooltip ARIA issue. Thoughts?

**Comment by bvaughn** [comment]
This repository is being merged into the main React repo ([github.com/facebook/react](https://github.com/facebook/react/)). As part of this, I am moving all issues to that repository as well and closing them here.

This issue has been relocated to:
https://github.com/facebook/react/issues/16490
